### ISSUE TITLE
Final Exams Card on Dashboard

### ISSUE THREAD

**Issue by roberthouse54** [mitodl/micromasters#2198]
This issue will cover all the different states of the Final Exams card on the dashboard. The card will be located at the top of the left column.

**1. After user has signed up, but before Pearson Profile has been created:**

![image](https://cloud.githubusercontent.com/assets/20047260/21433173/c95731ac-c83c-11e6-8054-89f6a187ed30.png)

On initial sign up, the card will contain information about the test, with a “Learn More” link, which links to the DEDP FAQs about Testing. We can also link to the Pearson Test Center search feature. The card will also show a message that “We will notify you when you are ready to schedule your exams.”

**2. After Pearson Test Account has been created, but user is not yet eligible to schedule any exams.**

![image](https://cloud.githubusercontent.com/assets/20047260/21433233/0a413bf4-c83d-11e6-85b0-cc0a9d5f8cab.png)
Once the user has a Pearson profile, we will show them their account info with a link to Edit their profile information if it is incorrect for whatever reason. Clicking the Edit icon will open the Edit Personal Info modal. We will continue to display the general information about the test with Learn More link.

If the user has updated their profile information by clicking the edit link, we will show them a message that their updates have been submitted, and to check back later.

![image](https://cloud.githubusercontent.com/assets/20047260/21433417/be320e90-c83d-11e6-9f8f-dcacd52beef9.png)

**3. For current users who do not have the required Pearson profile information**

If we need additional information for the Pearson profile (for current users) we will display a message that they need to update their profile.

![image](https://cloud.githubusercontent.com/assets/20047260/21433479/ef5ed778-c83d-11e6-9e29-5cefc7ed4e75.png)

**4. Profile is created and user is ready to schedule exam**

![image](https://cloud.githubusercontent.com/assets/20047260/21433526/26368d4a-c83e-11e6-8ecb-145cf0bcd543.png)

If the user is authorized to schedule an exam (or multiple exams), we will list any Courses for which they may schedule an exam. We will also have a button to “Schedule an Exam Now”. Clicking the button will take the user to Pearson to schedule the exam(s).

**Comment by rhysyngsun** [comment]
The learner is ready to register when two conditions are true:

```
from exams.models import ExamProfile, ExamAuthorization

ExamProfile.objects.filter(profile=profile, status=ExamProfile.PROFILE_SUCCESS).exists() # profile is valid and synced

# exams authorizations that synced successfully and are still valid
auths = ExamAuthorization.objects.filter(user=user, status=ExamAuthorization.STATUS_SUCCESS, date_last_eligible__lt=now)
```

If you expose these models via drf it's important that they're read-only since their lifecycle is managed on the backend.

**Comment by aliceriot** [comment]
Probably I will just add a read only field to the dashboard API for that

**Comment by pdpinch** [comment]
Is this in progress?

**Comment by aliceriot** [comment]
Yes

**Comment by pdpinch** [comment]
This can be closed by #2407 and #2408 are both closed.

**Comment by aliceriot** [comment]
I think we're read to close this now!
